### DISCUSSION TITLE
manifold mesh sampling

### DISCUSSION THREAD

**Discussion by xarthurx** [libigl/libigl#1859] [Ideas]
It is a bit unexpected to realize that `libigl` seems does not have a sampling function for 3D volume.

Similar to this one:
https://mfxvtk.readthedocs.io/en/latest/effects/sample-points-volume.html

![image](https://user-images.githubusercontent.com/1921878/126504707-2208955a-f6e2-43f5-8df0-da3a8739db9f.png)

I can naively get the bounding box and make a regular grid and test in/out with the given watertight mesh, but I'm sure there are more efficient algorithms.

Or I can make tets of the mesh and use the centroid of each tet?

Isn't it something that is often used in geometry processing?  Would it be possible to add one from the developer side?

@alecjacobson

**Comment by wtholliday** [comment]
`fast_winding_number` will tell you if a point is inside or outside a mesh
